numba/np/ufunc/array_exprs.py

```diff
@@ -77,7 +77,14 @@ def _match_array_expr(self, instr, expr, target_name):
         if ((expr_op in ('unary', 'binop')) and (
                 expr.fn in npydecl.supported_array_operators)):
             # It is an array operator that maps to a ufunc.
-            array_assigns[target_name] = instr
+            # check that all args have internal types
+            all_internal = True
+            for var in expr.list_vars():
+                if not self.typemap[var.name].is_internal:
+                    all_internal = False
+                    break
+            if all_internal:
```

Comment:
```suggestion
            # check that all args have internal types
            if all(self.typemap[var.name].is_internal
                   for var in expr.list_vars()):
```
think this is equivalent?

Reply:
@stuartarchibald 
done
thanks